server/courselist.rb

```diff
@@ -767,13 +767,11 @@ def self.getURLUniqname(url)
     # if not found say so.
     halt 404 if result.nil?
 
-    # if returning a file set the type type explicitly.
-    unless file_name.nil?
-      # by default it is text.
-      content_type 'text/plain'
-      # We also recognize png files.
-      content_type 'image/png' if file_name =~ /\.png$/
-    end
+    # get the return content type based on the file extension.
+    file_name =~ /\.([^.]+)$/
```

Comment:
Usually regular expressions are slow.  You might want to search backwards from the end of the string and find the last "." that demarcates filename from extension.